### ISSUE TITLE
GitHub application fails to authenticate.

### ISSUE THREAD

**Issue by H1X4Dev** [GrahamCampbell/Laravel-GitHub#109]
If you are using application method (creating application in https://github.com/settings/apps) to authenticate into GitHub API, you will be greeted with "Bad Credentials" error.

This issue is related to https://github.com/GrahamCampbell/Laravel-GitHub/pull/104 merge, as I have reverted it on my localhost and it works perfectly fine.

app/config.php
```
'default' => 'issues',
'connections' => [
    'issues' => [
        'method'       => 'application',
        'clientId'     => env('GITHUB_ISSUES_CLIENT_ID'    , null),
        'clientSecret' => env('GITHUB_ISSUES_CLIENT_SECRET', null),
    ],
]
```

Example usage:
```
GitHub::issues()->create('your-username', 'your-repository', [ 'title' => 'My Issue', 'body' => 'Hello World!' ]);
```

**Comment by H1X4Dev** [comment]
I guess, I'm reopening this. For anyone wondering, I have this resolved by simply switching to using a private key authentication (you can get it at the bottom), however.

When making an installation access token using:
```php
GitHub::apps()->createInstallationToken(config('github.connections.issues.installation_id'))
```

You will receive the following:
```
[2020-06-25 01:53:01] local.ERROR: 'Expiration time' claim ('exp') is too far in the future {"exception":"[object] (Github\\Exception\\RuntimeException(code: 401): 'Expiration time' claim ('exp') is too far in the future at .......\\vendor\\knplabs\\github-api\\lib\\Github\\HttpClient\\Plugin\\GithubExceptionThrower.php:105)
```

Seems like the issue exists in other projects, such as https://github.com/probot/probot/issues/942 and the simple fix for it would be to simply subtract one second off the expires at.

https://github.com/GrahamCampbell/Laravel-GitHub/blob/master/src/Auth/Authenticator/PrivateKeyAuthenticator.php#L50 :
```php
->expiresAt((new DateTimeImmutable('+10 minutes'))->getTimestamp() - 1)
```

After manually modifying it within the vendor, token installation token is created perfectly fine.
```
array:4 [
  "token" => "your-token-here"
  "expires_at" => "2020-06-25T03:05:01Z"
  "permissions" => array:2 [
    "issues" => "write"
    "metadata" => "read"
  ]
  "repository_selection" => "selected"
]
```

**Comment by GrahamCampbell** [comment]
Fixed in [v8.8.1](https://github.com/GrahamCampbell/Laravel-GitHub/releases/tag/v8.8.1) and [v9.2.1](https://github.com/GrahamCampbell/Laravel-GitHub/releases/tag/v9.2.1). 🚀
